paywall/src/data-iframe/Mailbox.ts

```diff
@@ -314,9 +318,11 @@ export default class Mailbox {
     this.postMessage(PostMessages.UPDATE_NETWORK, dataToSend.network)
     this.postMessage(PostMessages.UPDATE_LOCKS, dataToSend.locks)
     const unlockedLocks = this.getUnlockedLockAddresses()
+
     if (unlockedLocks.length) {
       this.postMessage(PostMessages.UNLOCKED, unlockedLocks)
-    } else {
+    } else if (Object.keys(dataToSend.locks).length) {
+      // Only send LOCKED if we actually have locks
```

Comment:
But is that enough? Could there be a case when we have a lock but have not retrieved the corresponding key yet?

Reply:
That case is certainly possible. I think as it currently stands, we don't have a way to know whether we have actually gotten all the keys. This change fixes the case that I was able to reproduce, and I think anything more thorough will require bigger changes to the design.